js/buttons.js

```diff
@@ -0,0 +1,103 @@
+function addButtons() {
+  var buttonBox = waterUseViz.elements.buttonBox = svg.append('g')
+    .attr('id', 'button-container');
+  
+  buttonBox.append('rect')
+    .attr('id', 'button-background')
+    .attr('height', waterUseViz.dims.buttonBox.heightDesktop);
+  
+  var buttonY = d3.scaleBand()
+    .domain(['title'].concat(categories))
+    .range([0, waterUseViz.dims.buttonBox.heightDesktop])
+    .paddingInner(0.1);
+  
+  // one group for each button
+  var buttons = buttonBox.selectAll('rect.button')
+    .data(categories)
+    .enter()
+    .append('g')
+    .classed('button', true)
+    .attr('transform', function(d) {
+      return 'translate(' + 10 + ', ' + buttonY(d) + ')';
+    });
+  
+  // button rectangles
+  buttons.append('rect')
+    .attr('x', 0)
+    .attr('y', 0)
+    .attr('height', buttonY.bandwidth()) // width gets handled in updateButtons
+    .on('click', function(d){
+      updateCategory(d.toLowerCase(), activeCategory);
+    })
+    .on('mouseover', function(d){
+      showCategory(d.toLowerCase(), activeCategory, action = 'mouseover');
+    })
+    .on('mouseout', function(d){
+      showCategory(activeCategory, d.toLowerCase(), action = 'mouseout');
+    });
+  
+  // button category labels
+  buttons.append('text')
+    .classed('category-label', true)
+    .text(function(d){
+      return categoryToName(d);
+    })
+    .attr('y', function(d) {
+      return buttonY.bandwidth()/2;
+    });
+
+  // button category use values
+  buttons.append('text')
+    .classed('category-amount', true)
+    .text(function(d){
+      return 'XX';
+    })
+    .style('font-weight', 'thick')
+    .attr('text-anchor', 'end')
+    .attr('y', function(d) {
+      return buttonY.bandwidth()/2;
+    });
+  
+  // add styling to the buttons and text according to which is active
+  updateButtons(activeCategory);
+  
+}
+
+function updateButtons(category) {
+  waterUseViz.elements.buttonBox.selectAll('.button rect')
+    .style('fill', function(d) {
+      if(d === category) {
+        return categoryToColor(d);
+      } else {
+        return '#c6c6c6';
+      }
+    })
+    .attr('width', function(d) {
+      if(d === category) {
+        return waterUseViz.dims.buttonBox.width * 0.95;
+      } else {
+        return waterUseViz.dims.buttonBox.width * 0.7;
+      }
+    });
+  
+  waterUseViz.elements.buttonBox.selectAll('.button .category-label')
```

Comment:
what if we put these styles into a class called `button-selected` or something, and the if else was in classed here
```
.classed("selected", function(d) {
     if(d === category) {
          return true;
     } else {
          return false;
     }
});
```

Reply:
i played with some options for that. one challenge is that the selected button has a different color from any other selected button, so a `selected` class for css would have to omit color, and width is calculated (and i'd like it to be transitioned) and so would also need to be a d3 thing. i also tried defining the default color in one css definition and the specific colors in other css/d3 locations, but i couldn't get the priorities right in my first effort. i'll take one more shot at using more css for this, at least for a subset of the properties.